packages/core/saved-objects/core-saved-objects-migration-server-internal/src/actions/update_mappings.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import * as Either from 'fp-ts/lib/Either';
+import * as TaskEither from 'fp-ts/lib/TaskEither';
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import type { IndexMapping } from '@kbn/core-saved-objects-base-server-internal';
+import { catchRetryableEsClientErrors } from './catch_retryable_es_client_errors';
+import type { RetryableEsClientError } from './catch_retryable_es_client_errors';
+import { DEFAULT_TIMEOUT } from './constants';
+
+/** @internal */
+export interface UpdateMappingsParams {
+  client: ElasticsearchClient;
+  index: string;
+  mappings: IndexMapping;
+}
+/**
+ * Updates an index's mappings and runs an pickupUpdatedMappings task so that the mapping
+ * changes are "picked up". Returns a taskId to track progress.
+ */
+export const updateMappings = ({
+  client,
+  index,
+  mappings,
+}: UpdateMappingsParams): TaskEither.TaskEither<
+  RetryableEsClientError | 'incompatible_mapping_exception',
+  'update_mappings_succeeded'
+> => {
+  return () => {
+    return client.indices
+      .putMapping({
+        index,
+        timeout: DEFAULT_TIMEOUT,
+        ...mappings,
+      })
+      .then(() => Either.right('update_mappings_succeeded' as const))
+      .catch(catchRetryableEsClientErrors)
+      .catch(() => Either.left('incompatible_mapping_exception'));
```

Comment:
Just for my learning (and not part of this PRs changes):

is it possible that ES returns some other kind of error code here that we are mapping to `incompatible_mapping_exception`. I see the following for an incompatible mapping on a recent snapshot:

```
{
	"error": {
		"root_cause": [
			{
				"type": "illegal_argument_exception",
				"reason": "mapper [file.Status] cannot be changed from type [keyword] to [long]"
			}
		],
		"type": "illegal_argument_exception",
		"reason": "mapper [file.Status] cannot be changed from type [keyword] to [long]"
	},
	"status": 400
}
```

I'm not sure if there are others, didn't spot 'em. Might be more defensive to check for the `type: 'illegal_argument_exception'` and just throw for an unexpected condition.

Reply:
I was fixing the CI errors and came to the same conclusion, this will be part of my next push.
TY!